crates/release_plz_core/src/command/release.rs

```diff
@@ -1040,19 +1061,42 @@ mod tests {
         let token = "t0p$eCrEt";
         let token_env_var = format!("CARGO_REGISTRIES_{}_TOKEN", registry_name.to_uppercase());
 
-        let old_value = env::var(&token_env_var);
-        env::set_var(&token_env_var, token);
+        with_env_var(&token_env_var, token, || {
+            let request = ReleaseRequest::new(fake_metadata()).with_registry(registry_name);
+            let registry_token = request.find_registry_token(Some(registry_name)).unwrap();
 
-        let request = ReleaseRequest::new(fake_metadata()).with_registry(registry_name);
-        let registry_token = request.find_registry_token(Some(registry_name)).unwrap();
+            assert!(registry_token.is_some());
+            assert_eq!(token, registry_token.unwrap().expose_secret());
+        });
+    }
 
-        if let Ok(old) = old_value {
-            env::set_var(&token_env_var, old);
-        } else {
-            env::remove_var(&token_env_var);
-        }
+    #[test]
+    fn should_reference_env_var_provided_index() {
+        use cargo_utils::registry_url;
+
+        let _guard = NO_PARALLEL.lock().unwrap();
```

Comment:
should we acquire the lock inside the `with_env_var` function?
I think that's the critical part of the code we can't run in parallel